jac-uk/admin
# #2073 Allow Users To Review Exercise Changes Prior To Approval

## What's included?
When a user unlocks an exercise to edit it, ensure the subsequent approver can see a list of the fields that have changed.

**This work is linked to the following hotfix PR in Digital Platform**: https://github.com/jac-uk/digital-platform/pull/926
**This work is linked to the following PR in Jac Kit**: https://github.com/jac-uk/jac-kit/pull/142

Closes #2073 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
The fields that need testing for changes are the following:

aboutTheRole,
aboutTheRoleWelsh,
additionalWorkingPreferences,
advertType,
applicationCloseDate,
applicationOpenDate,
appliedSchedule,
appointmentType,
aSCApply,
assessmentMethods,
assessmentOptions,
authorisations,
characterChecks,
contactIndependentAssessors,
criticalAnalysisTestDate,
criticalAnalysisTestEndTime,
criticalAnalysisTestStartTime,
downloads,
estimatedLaunchDate,
exerciseMailbox,
exercisePhoneNumber,
feePaidFee,
finalOutcome,
futureStart,
immediateStart,
independentAssessmentsReturnDate,
inviteOnly,
isCourtOrTribunal,
isSPTWOffered,
jurisdiction,
jurisdictionQuestion,
jurisdictionQuestionAnswers,
jurisdictionQuestionType,
location,
locationQuestion,
locationQuestionAnswers,
locationQuestionType,
memberships,
name,
noSalaryDetails,
otherJurisdiction,
otherLOS,
otherMemberships,
otherQualifications,
otherRetirement,
otherShortlistingMethod,
otherYears,
pjeDays,
postQualificationExperience,
previousJudicialExperienceApply,
qualifications,
reasonableLengthService,
referenceNumber,
retirementAge,
roleSummary,
roleSummaryWelsh,
salary,
salaryGrouping,
scenarioTestDate,
scenarioTestEndTime,
scenarioTestStartTime,
schedule2Apply,
selectionCriteria,
selectionDays,
selectionExerciseManagerFullName,
shortlistingMethods,
siftStartDate,
siftEndDate,
nameBlindSiftStartDate,
nameBlindSiftEndDate,
shortlistingOutcomeDate,
situationalJudgementTestDate,
situationalJudgementTestEndTime,
situationalJudgementTestStartTime,
subscriberAlertsUrl,
typeOfExercise,
uploadedCandidateAssessmentFormTemplate,
uploadedIndependentAssessorTemplate,
uploadedJobDescriptionTemplate,
uploadedTermsAndConditionsTemplate,
welshPosts,
welshRequirement,
welshRequirementType,
yesSalaryDetails

Use this link to test: https://jac-admin-develop--pr2088-feature-2073-allow-u-67azf0zb.web.app/exercise/RbLNndN78COCkwapBkC2/details/overview

For each test below it is convenient to have two separate browsers open.
I have added a **screen recording below** that illustrates this.
The browsers should be different so:

-  choose two browsers from (chrome, safari, edge etc) and open a window for each
- put the windows side-by-side
- in browser 1 you should login as JAC Staff
- in browser 2 you should login as a Senior Leader.

### TEST 1
- Select an existing live exercise
- Click to Unlock the exercise for editing
- Make some changes
- Click Submit For Approval on the Overview page
- The approver should see a message above the Approve/Reject buttons showing them the fields that have changed

### TEST 2
- Create a new exercise
- Fill in all the sections so that the Submit For Approval button is displayed on the Overview page
- Submit it for approval
- Approve it
- Publish it
- Click to Unlock the exercise for editing
- Make some changes
- Click Submit For Approval on the Overview page
- The approver should see a message above the Approve/Reject buttons showing them the fields that have changed

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
See screen recording:

https://github.com/jac-uk/admin/assets/115651787/e7fd8208-d4c3-4790-8f3d-201cec5e46ab

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
